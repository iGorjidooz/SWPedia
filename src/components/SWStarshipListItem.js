import React from 'react';
import { Link } from 'react-router-dom';

const SWStarshipListItem = ({ starship, listIndex }) => {
   return (
      <Link to={`/starship/${starship.id}`}>
         <h3>{listIndex} {starship.name}</h3>
         <p>model: {starship.model}</p>
         <p>manufacturer: {starship.manufacturer}</p>
      </Link>
   );
};

export default SWStarshipListItem;