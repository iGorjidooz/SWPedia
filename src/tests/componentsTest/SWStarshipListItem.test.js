import React from 'react';
import { shallow } from 'enzyme';
import starshipFixtures from '../fixtures/starshipsFixtures';
import SWStarshipListItem from '../../components/SWStarshipListItem';

 test('should render SWCharacterListItem correctly', () => {
    const wrapper = shallow(
      <SWStarshipListItem starship={starshipFixtures[1]} listIndex={starshipFixtures[1].id} />
    );
    expect(wrapper).toMatchSnapshot();
 })