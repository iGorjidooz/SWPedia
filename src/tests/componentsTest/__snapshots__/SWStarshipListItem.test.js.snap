// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SWCharacterListItem correctly 1`] = `
<Link
  to="/starship/3"
>
  <h3>
    3
     
    Star Destroyer
  </h3>
  <p>
    model: 
    Imperial I-class Star Destroyer
  </p>
  <p>
    manufacturer: 
    Kuat Drive Yards
  </p>
</Link>
`;
